akvo/rsr/management/commands/migrate_related_projects.py

```diff
@@ -0,0 +1,223 @@
+            sibling_group
+            for sibling_group in sibling_uuid_groups
+            if project_uuid in sibling_group or sibling_uuid in sibling_group
+        ]
+        group_count = len(groups)
+        if group_count == 0:
+            # New group
+            sibling_uuid_groups.append({project_uuid, sibling_uuid})
+        elif group_count == 1:
+            # Add to existing group
+            groups[0].add(project_uuid)
+            groups[0].add(sibling_uuid)
+        else:
+            # Multiple groups should be merged into one
+            first_group = groups[0]
+            groups_to_merge = groups[1:]
+            first_group.update(*groups_to_merge)
+            # Remove merged groups
+            for group in groups_to_merge:
+                groups.remove(group)
+    return sibling_uuid_groups
+
+
+def resolve_sibling_groups(
+        related_siblings: List[RelatedProject],
+        sibling_uuid_groups: List[Set[UUID]]
+) -> List[List[Project]]:
+    """Use cache to resolve project UUIDs in the groups to projects"""
+    sibling_projects_cache: Dict[UUID, Project] = {}
+    for related_sibling in related_siblings:
+        sibling_projects_cache[related_sibling.project.uuid] = related_sibling.project
+        sibling_projects_cache[related_sibling.related_project.uuid] = related_sibling.related_project
+    sibling_groups: List[List[Project]] = [
+        [sibling_projects_cache[sibling_id] for sibling_id in sibling_id_group]
+        for sibling_id_group in sibling_uuid_groups
+    ]
+    return sibling_groups
+
+
+def set_sibling_parents(sibling_groups: List[List[Project]]) -> List[Project]:
+    """
+    Attempts to sets the parent of each group of sibling projects
+
+    Not all groups will have the same parent (or a parent at all), which is of course a problem.
+
+    :return: All the projects that now have a parent
+    """
+    modified_projects = []
+    orphaned_siblings = []
+    multi_parent_siblings = []
+    for sibling_group in sibling_groups:
+        # Find parents
+        parents = {}
+        for sibling in sibling_group:
+            parent_uuid = sibling.get_parent_uuid()
+            if parent_uuid:
+                parents.setdefault(parent_uuid, []).append(sibling)
+
+        #
+        parent_count = len(parents)
+        if parent_count == 0:
+            # print(f"{sibling_group} are all orphans")
+            orphaned_siblings.append(sibling_group)
+        elif parent_count == 1:
+            parent_uuid, _ = parents.popitem()
+            parent = Project.objects.get(uuid=parent_uuid)
+            for sibling in sibling_group:
+                if not sibling.get_parent_uuid():
+                    sibling.set_parent(parent, True)
+                    modified_projects.append(sibling)
+            print(f"f{parent_uuid} is the parent of {sibling_group}")
+        else:
+            print(f"{sibling_group} has multiple parents!")
+            multi_parent_siblings.append(sibling_group)
```

Comment:
The `multi_parent_siblings` variable is not used for anything, It should be omitted IMO.

Reply:
Fixed in 8d4c43e08535895f4039163e46e03b59c130b85f